src/metatensor/models/cli/train.py

```diff
@@ -284,6 +283,16 @@ def train_model(
     ###########################
     logger.info("Setting up model")
 
+    # TODO: A more direct way to look up the gradients would be to get them from
+    # the configuration dict of the training run.
+    gradients: Dict[str, List[str]] = {}
+    for train_options in train_options_list:
+        for key in train_options["targets"].keys():
+            # look inside training sets and find gradients
+            for train_dataset in train_datasets:
+                if key in train_dataset[0].keys():
```

Comment:
You are only considering `train_dataset[0]`. What if there are more datasets.

Reply:
`train_dataset[0]` is taking the first sample of that dataset, which is guaranteed to be the same as all the others from that dataset. Different datasets are taken into account with the second loop (`for train_dataset in train_datasets`)

Reply:
It seems like this code was a leftover from another PR, it's gone now